### ISSUE TITLE
Installing any distro fails with "WslRegisterDistribution failed with error: 0x80070002 Error: 0x80070002 The system cannot find the file specified."

### ISSUE THREAD

**Issue by k3rst1n** [microsoft/WSL#4959]
Hi,

After installing Windows 10 on my desktop, I installed the Debian distribution. Right after installation, it loaded and worked fine, however after closing the shell and re-opening it, it got stuck while loading and my CPU usage increased to 100% - in the services I could see around 8 processed called "init" that were running and using up all cpu resources.
I wasn't able to kill the processes, only a reboot helped. As the processes came back every time I opened the Debian shell, I eventually deinstalled the disto.
Now, when I try to install any distro, I always end up with the same error:

```
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80070002
Error: 0x80070002 The system cannot find the file specified.

Press any key to continue...
```
I tried all possible combinations of deinstalling, reboot, etc.
 wsllist never shows any of the Distros, even if they show up in the installed programs.

```
wsl --list --all
Windows Subsystem for Linux has no installed distributions.
Distributions can be installed by visiting the Microsoft Store:
https://aka.ms/wslstore
```
Attached you can find my logs - I was collecting them while I tried to install the Ubuntu distribution.

[logs.zip](https://github.com/microsoft/WSL/files/4301868/logs.zip)

Windows version: Win 10 Pro, Version 1909 Build 18363.657.

Please let me know if you require anything else!

Thank you,
Kerstin

**Comment by CBornes** [comment]
Same here. I add this working. Today I've tried to open my Ubuntu terminal and I'm having the same problem.

**Comment by spaceraccoon** [comment]
Same here. Happened when I upgraded to WSL 2.

**Comment by spaceraccoon** [comment]
@CBornes @k3rst1n Did yall find any fixes for this?

**Comment by k3rst1n** [comment]
@spaceraccoon No unfortunately not - still not working

**Comment by rappster** [comment]
Same here

**Comment by NicoWde** [comment]
I also have this issue on 2004 Build 20150.1000

**Comment by donxp** [comment]
Same issue here on Windows version 2004 build 19041.329

**Comment by chancez** [comment]
I was able to get this resolved by removing the WSL kernel update from programs/features, and then removing the WSL feature, rebooting, and then re-enabling the WSL feature and rebooting. I also deleted my WSL instances during this process.

**Comment by NicoWde** [comment]
> 
> 
> I was able to get this resolved by removing the WSL kernel update from programs/features, and then removing the WSL feature, rebooting, and then re-enabling the WSL feature and rebooting. I also deleted my WSL instances during this process.

Hey @chancez, I tried this at 20150.1000 - it unfortunatly did not help.

**Comment by simmessa** [comment]
> I was able to get this resolved by removing the WSL kernel update from programs/features, and then removing the WSL feature, rebooting, and then re-enabling the WSL feature and rebooting. I also deleted my WSL instances during this process.

Thank You for the info, I've tried this partially, by uninstalling that WSL kernel update because since a couple of reboots ago WSL2 is completely broken... I'm desperate to recover my WSL2 filesystem, hopefully somebody from the dev team can help us here :(

**Comment by simmessa** [comment]
I'm on build #19592 and my WSL2 only says this:
![image](https://user-images.githubusercontent.com/1277216/85272733-daeb0080-b47c-11ea-8928-33763796467e.png)

I've tried de-registering Ubuntu but it's dead and won't launch... I'm really in need to recover the WSL2 filesystem if there's a way. oh, and btw to make it work again!

Thanks for your time / help.

**Comment by alminisl** [comment]
The only way I fixed this is by disabling WSL and re-enabling it again, I tried everything.

**Comment by simmessa** [comment]
> The only way I fixed this is by disabling WSL and re-enabling it again, I tried everything.

WSL2 on insider previews has been so unreliable I've gone back to something I can control: an Ubuntu VM made in HyperV which is basically the same, and I'm already feeling the experience is much better.

I now feel confident that no windows update will wipe out my WSL2 distros, as a next step I'm opting out of insider previews, I'm probably done with WSL and evangelizing it, It's a good concept but the execution is so lacking... I don't feel it's really worth it.

**Comment by Picoseconds** [comment]
> insider previews

> unreliable

Well, it's not like you weren't warned about it. I don't believe this is a valid reason to hate on WSL2.

**Comment by simmessa** [comment]
Sure, i was aware of the insider "disclaimer" and that's just my personal opinion.

But I won't deny I'm sick of WSL2, too many things broke at the wrong time, I'm outta here. Enjoy the ride!

**Comment by danvln** [comment]
Bumping this since it is really an issue and blocked the usage of WSL 2

**Comment by JosXa** [comment]
Issue still appears in December.

**Comment by nagarajcruze** [comment]
have anyone found fix for this issue? i was using 2sl2 kali-linux peacefully but today i tried to overwrite the kali-linux.vhdx file, then this problem started for me and i did every fixes posted on other git issues, still i can't find any fix. anyone help me .

**Comment by enixam** [comment]
the same issue. I tried reinstalling wsl and disabling virtual machine and then enabling it again, didn't work.

**Comment by CollapsedMetal** [comment]
Same problem here! 

As people are suggesting on #5107, I also think that it is related to the absence of `vEthernet (WSL) adapter`. No matter how much I try reinstalling WSL stuff in different order, there is no way the nic adapter being installed again.

Iv'e also tried to "manually" install the adapter via Hyper-V with no luck.😥

**Comment by benyaminl** [comment]
> 
> 
> Same problem here!
> 
> As people are suggesting on #5107, I also think that it is related to the absence of `vEthernet (WSL) adapter`. No matter how much I try reinstalling WSL stuff in different order, there is no way the nic adapter being installed again.
> 
> Iv'e also tried to "manually" install the adapter via Hyper-V with no luck.😥

I Agree, I also having this problem on 20H2, it's frustating

**Comment by WassimBenzarti** [comment]
I encountered this issue today. I had Docker Desktop installed, I removed it but no luck starting with WSL Ubuntu :(

**Comment by jyjwhbk** [comment]
> I encountered this issue today. I had Docker Desktop installed, I removed it but no luck starting with WSL Ubuntu :(

exactly same situation as u, and have no idea what to do

**Comment by Forbzy** [comment]
I wish I hadn't of updated. Everything was working fine util about a week ago. I hope we are getting close to a fix reliable permanent fix.

**Comment by amchii** [comment]
Same

**Comment by Dzivo** [comment]
Same here

**Comment by robertoti** [comment]
I also have this issue on 21H1 Build 19043.1081

**Comment by mattkotys** [comment]
Me too.

**Comment by abdilahrf** [comment]
same ;(

**Comment by Haebuk** [comment]
same..

**Comment by amirwesthoff** [comment]
same here

**Comment by merskip** [comment]
I have the same issue. OS Version 10.0.19043 N/A Build 19043.

**Comment by er-eis** [comment]
Same issue here on `Microsoft Windows [Version 10.0.22000.160]`

**Comment by dot1209** [comment]
I have the same issue on 21H1 Build 19043.1237.
But I tried `wsl --set-default-version 1` and this works for me.

**Comment by Zeratoxx** [comment]
> I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.

I guess that fixed it for me!

**Comment by averyfreeman** [comment]
I was switching to WPM/WinGet from Chocolatey because I have a bunch of software installed through `choco` that didn't install shortcuts, thinking `winget` would be more reliable - then this happened 😂

**Comment by bmasterc** [comment]
Kaspersky Antivirus disabled WSL from starting when I unknowingly told it to fix running Linux apps from Windows. To enable, lookup how to set LxssManager startup to manual via registry (Access Denied on the GUI)

**Comment by peterburgs** [comment]
> The only way I fixed this is by disabling WSL and re-enabling it again, I tried everything.

I double tried, triple tried but seems like your solution did not work

**Comment by hguitar** [comment]
for anyone still looking for solution here, try to delete the .wslconfig under your C:\Users\<your username>\.

**Comment by gibait** [comment]
> for anyone still looking for solution here, try to delete the .wslconfig under your C:\Users<your username>.

I uninstalled Docker Desktop. Removed Ubuntu install. Disabled and re-enabled WSL. This is what made it work eventually! Thank you

**Comment by Durbek-Gafur** [comment]
> I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> 
> Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> 
> I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> 
> After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)

Using CCleaner and cleaning registry helped me. Thanks

**Comment by copapp** [comment]
help, couldnt solve by solution above, I have tried

- docker desktop was uninstalled
- I dont have .wslconfig in `C:\Users\<username>\` 
- wsl feature off->ccleaner to clean registry->reboot->wsl feature on again->reboot-> `wsl --install -d Ubuntu` return `WslRegisterDistribution failed with error: 0x80070002`

I thought it was because `wuauserv` was stopped, but wsl --install was run with `wuauserv` running. I dont know what else I could try

**Comment by EricGT** [comment]
Could not find root cause of problem.

What achieved a successful install for me was to forgo using winget and just install from Microsoft Store.

Just to see if that cleared the root cause, used `wsl --unregister`, rebooted, then tried

> winget install -e --id Canonical.Ubuntu

which resulted in

`Installer failed with exit code: 0x8007002 : The system cannot find the file specified.`

So maybe this is a `winget` issue and not a `WSL` issue.

**Comment by Stormholt** [comment]
> Same problem here!
> 
> As people are suggesting on #5107, I also think that it is related to the absence of `vEthernet (WSL) adapter`. No matter how much I try reinstalling WSL stuff in different order, there is no way the nic adapter being installed again.
> 
> Iv'e also tried to "manually" install the adapter via Hyper-V with no luck.😥

My solution based on CollapsedMetals idea that the problem can be related to missing the `vEthernet (WSL) adapter`. Was to try and fix its as a [missing network adapter](https://www.makeuseof.com/windows-10-network-adapter-missing/). When i made it to windows network reset, what ended up happening was that all my network adapters got corrupted and returned code 56 forever. This error led me to [this](https://answers.microsoft.com/en-us/windows/forum/all/code-56-on-every-device-after-network-reset/69577a08-e173-4d9c-bbc7-2066e1dd905c), which made me update to the latest windows from a USB drive. After the update my internet adapters worked fine, `vEthernet (WSL) adapter` had returned, and WSL2 worked fine after installing the latest kernel update.

**Comment by himansh-u** [comment]
wsl --set-default-version 1
**seems to work on build 2004**

**Comment by Faniche** [comment]
I got same problem. And I remembered that I uninstalled the VMware Workstation Player before the problem occur. So I tried to reinstall the VMware Workstation Player and it was solved. You could try to install VMware Workstation Player if it helps.

**Comment by handmade-medium** [comment]
> for anyone still looking for solution here, try to delete the .wslconfig under your `C:\Users\<your username>\`.

This worked for me right away, no reboots or uninstall/reinstall required, thank you!

**Comment by sumitNITS** [comment]
Error: 0x80070002 The system cannot find the file specified
![error](https://user-images.githubusercontent.com/37767537/224254625-62b2eeb5-0ba1-4936-8e92-42db4f822af4.png)

Here's how I fixed it today! Thanks, everyone for this wonderful discussion above, it did help me to fix it!

1. Install the free version of CCleaner and use it to fix the issues present in your system.
I've used v6.09 at the time of doing
![CCleaner](https://user-images.githubusercontent.com/37767537/224257730-35c29823-18c9-4117-af11-de5da1222698.png)

2. Check whether Windows feature windows subsystem for Linux is on. If not then turn it ON and restart your machine
![Windows feature](https://user-images.githubusercontent.com/37767537/224257941-08404c2a-2e51-4e2e-b9a3-18127613b559.png)

3. From the windows store get your desired Ubuntu version and run it. Set your username and password. That's all. Now it will work.
![solved](https://user-images.githubusercontent.com/37767537/224258432-d31a4dee-d3e3-4bf7-ad9d-31038327c169.png)

**Comment by Peugin** [comment]
I got the problem when OS upgrade from 20H2 to 21H2, and I upgrade again to 22H2 build 19045.2846 it fixed.

**Comment by Erokos** [comment]
> I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> 
> Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> 
> I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> 
> After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)

Thank you, from all the proposed solutions only this one worked for me! You're a life saver!

**Comment by nasNasa** [comment]
> I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> 
> Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> 
> I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> 
> After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)

i wanna just say thanks for your sharing. it help me to solve my mind blowing problem.

**Comment by Francis-Ferri-personal** [comment]
> I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> 
> Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> 
> I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> 
> After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)

I just had to use CCleaner to remove the registry

**Comment by Li-can-cheng** [comment]
注册表问题，用CCcleaner解决。

**Comment by Leonezz** [comment]
Same issue suddenly occurs, and none of the above solutions work for me.

**Comment by ecotner** [comment]
`wsl --set-default-version 1` seemed to "work" temporarily for me, but ended up creating a new distribution with none of my existing files. So I eventually had to wipe WSL (not just the distro, like all of WSL) and reinstall it. Make sure to back up your filesystem before deleting/uninstalling anything by backing up `C://Users/<your user>/AppData/Local/Packages/<your distro>/LocalState/ext4.vhdx`, then copying the backup to the original location once you're reinstalled everything. In my case (using the `Ubuntu-20.04` distro) the `<your distro>` folder was called `CanonicalGroupLimited.Ubuntu20.04LTS_79rhkp1fndgsc`.

**Comment by rz1027** [comment]
> I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> 
> Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> 
> I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> 
> After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)


Ok I have to confess, I thought I caused this problem for myself by blindly deleting huge vhdx's and uninstalling docker desktop.
But just before this problem I coincidentally installed VMware Workstation. 
It seems interestingly unrelated but solved my problem like charm, despite trying tons of other solution.
Cheers bro, you're a magician

**Comment by Definelymes** [comment]
I tired many things over past 24 hours, and the only thing that worked for me was removing the whole wsl and then re-installing it.

**Comment by hieudtrung** [comment]
> I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> 
> Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> 
> I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> 
> After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)

After years of development, this WSL2 issue still hit my computer today. I tried all solutions proposed here and installing CCleaner was the only way. Thank you very much.

**Comment by Cameron-Nguyen1** [comment]
> > I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> > Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> > I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> > After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)
> 
> After years of development, this WSL2 issue still hit my computer today. I tried all solutions proposed here and installing CCleaner was the only way. Thank you very much.

CCleaner -> Tools -> Registry -> Scan -> Apply Fix (Works for Windows 10 home edition ubuntu 22.04 in August, 2024)

Thanks for sharing, hieudtrung.

**Comment by msuzuki55** [comment]
I occurred 0x80070002 because C:\Program Files\WSL\system.vhd was not found.
I restored system.vhd using the steps below and the problem was resolved.
https://github.com/microsoft/WSL/issues/11069#issuecomment-2055653578

**Comment by Trimatix** [comment]
My company had deployed the latest version of the Microsoft Defender for Endpoint plugin for WSL via Intune. This failed to install on multiple PCs, causing this error when launching WSL:
> Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80070002
Error: 0x80070002 The system cannot find the file specified.

Or one of these errors when launching docker desktop:
>deploying WSL2 distributions
ensuring main distro is deployed: deploying "docker-desktop": importing WSL distro "A fatal error was returned by plug-in 'DefenderforEndpointPlug-in'\r\nError code: Wsl/Service/RegisterDistro/CreateVm/Plugin/ERROR_FILE_NOT_FOUND\r\n" output="docker-desktop": exit code: 4294967295: running WSL command wsl.exe C:\Windows\System32\wsl.exe --import docker-desktop <HOME>\AppData\Local\Docker\wsl\main C:\Program Files\Docker\Docker\resources\wsl\wsl-bootstrap.tar --version 2: A fatal error was returned by plug-in 'DefenderforEndpointPlug-in'
> 
> Error code: Wsl/Service/RegisterDistro/CreateVm/Plugin/ERROR_FILE_NOT_FOUND
> 
> : exit status 0xffffffff
> checking if isocache exists: CreateFile \\wsl$\docker-desktop-data\isocache\: The filename, directory name, or volume label syntax is incorrect.

Or:
> checking if isocache exists: CreateFile \\wsl$\docker-desktop-data\isocache\: The network name cannot be found.

I resolved the issue on computer by repairing the plugin installation from control panel, as described in the documentation: https://learn.microsoft.com/en-us/defender-endpoint/mde-plugin-wsl#installation-steps:~:text=If%20you%20see%20an%20error%20on%20launching%20WSL%2C%20such%20as%20%22A%20fatal%20error%20was%20returned%20by%20plugin%20%27DefenderforEndpointPlug%2Din%27%20Error%20code%3A%20Wsl/Service/CreateInstance/CreateVm/Plugin/ERROR_FILE_NOT_FOUND%22%2C%20it%20means%20the%20Defender%20for%20Endpoint%20plug%2Din%20for%20WSL%20installation%20is%20faulty.%20To%20repair%20it%2C%20follow%20these%20steps%3A

**Comment by velicaferov5** [comment]
Had same issue, tried a lot of suggestions, and was not resolved.
Installed [Microsoft Defender for Endpoint plug-in for Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/defender-endpoint/mde-plugin-wsl#installation-steps), followed steps and did troubleshooting in [microsoft-defender-for-endpoint](https://blog.sonnes.cloud/microsoft-defender-for-endpoint-plug-in-for-windows-subsystem-for-linux-wsl-finally/)
I hope will be also fixed for others.

**Comment by HugoRebelo99** [comment]
I was able to fix this issue by using `wsl.exe --set-default-version 1`, and then `wsl.exe --install <distro>`. After this it worked like a charm, I was able to install the distros but I was limited to version 1, every time I wanted to `wsl.exe --set-version <distro> 2` it gave me a different error. WSL version 1 doesn't allow me to run Docker from within WSL. After a long time and checking other issues, I tried to update my WSL kernel and even this was unsuccessful. In the end, the folder `C:\Windows\System32\lxss` didn't contain the `lib` folder and the `kernel` and `license` files under `C:\Windows\System32\lxss\tools`. I just grabbed those files from another WSL installation on another PC and got the issue fixed! 😁 
I don't know how those files went missing, but I assure you that disabling/enabling the Windows features such as Hyper-V, WSL, Windows Hypervisor Platform, and Virtual Machine Platform doesn't fix the issue nor bring these files back.

Hope this helps. 😉

TLDR: Check if you have `C:\Windows\System32\lxss\tools\(kernel/license)` files and if not try grabbing them from another WSL installation into your PC.

**Comment by AdrianDeWinter** [comment]
> I was able to fix this issue by using `wsl.exe --set-default-version 1`, and then `wsl.exe --install <distro>`. After this it worked like a charm, I was able to install the distros but I was limited to version 1, every time I wanted to `wsl.exe --set-version <distro> 2` it gave me a different error. WSL version 1 doesn't allow me to run Docker from within WSL. After a long time and checking other issues, I tried to update my WSL kernel and even this was unsuccessful. In the end, the folder `C:\Windows\System32\lxss` didn't contain the `lib` folder and the `kernel` and `license` files under `C:\Windows\System32\lxss\tools`. I just grabbed those files from another WSL installation on another PC and got the issue fixed! 😁 I don't know how those files went missing, but I assure you that disabling/enabling the Windows features such as Hyper-V, WSL, Windows Hypervisor Platform, and Virtual Machine Platform doesn't fix the issue nor bring these files back.
> 
> Hope this helps. 😉
> 
> TLDR: Check if you have `C:\Windows\System32\lxss\tools\(kernel/license)` files and if not try grabbing them from another WSL installation into your PC.

Oh my goodness, I have been trying everything I could think of for the past two weeks, thank you so much!
Just for SEO to help others get here, my problems began after I did an in-place upgrade of my server to Windows Server 2025

**Comment by vivek711chaudhary** [comment]
> I was able to get this resolved by removing the WSL kernel update from programs/features, and then removing the WSL feature, rebooting, and then re-enabling the WSL feature and rebooting. I also deleted my WSL instances during this process.

Thanks this worked for me. I disabled the wsl in the control panel and then re-enabled it

**Comment by gdmdxl** [comment]
Thanks this worked for me. 

> > I just managed to fix this, but am unsure of the exact underlying issue, or what caused it (the original issue was Docker Desktop failing to start with various random errors).
> > Firstly I noticed the comment above about a possible link to network adapters. I noticed I had a "VMWare Bridge Protocol" attached to my ethernet adapter, probably left over from an old VMWare Player install, but I was unable to remove/uninstall it (0x8007007E error).
> > I then did a Network Reset in Windows, which broke _all_ my networking, the adapters had the error "Windows is still setting up the class configuration for this device, Code 56" which is relatively obscure, and again apparently related to an incompatible underlying driver. This error seems to be impossible to solve by a user (ie. through the normal Windows UI) except by a system restore or something. I was only able to fix it by running the CCleaner software (free version), and accepting all it's recommendations in the registry clean section.
> > After a reboot all the networking worked fine again, and I was able to install WSL2 and then reinstall Docker Desktop (which is what started me on the trail of fixing it all!)
> 
> After years of development, this WSL2 issue still hit my computer today. I tried all solutions proposed here and installing CCleaner was the only way. Thank you very much.
